### PR TITLE
Fix #19 - Reduce the DeX Size and reduce the memory allocation size b…

### LINKED ISSUE #19: Reduce the DeX Size and reduce the memory allocation size by avoiding Enums
We have to replace _useless_ enums by TypeDef Constant.

More info about this here : https://android.jlelse.eu/android-performance-avoid-using-enum-on-android-326be0794dc3

### PR DESCRIPTION
**Pull request by vmichalak**
…y avoiding Enums

### CODE REVIEW

File: Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/enums/DataEntryStatus.java [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
+
+@IntDef({DataEntryStatus.ADDED, DataEntryStatus.DID_NOT_CHANGED, DataEntryStatus.DELETED})
+@Retention(RetentionPolicy.SOURCE)
+public @interface DataEntryStatus
 {
     /**
      * The data has been added locally and has not been synchronized yet.
      */
-    Added,
+    int ADDED = 1;
```
Comment (veler) [inline Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/enums/DataEntryStatus.java]: Attention, be sure that those numbers corresponds to what we have here : [ClipboardZanager.Core.Desktop/Enums/DataStatus.cs](https://github.com/veler/clipboardzanager/blob/master/ClipboardZanager/Sources/ClipboardZanager.Core.Desktop/Enums/DataStatus.cs)

File: Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/enums/ThumbnailDataType.java [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
+import android.support.annotation.IntDef;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+@IntDef({ThumbnailDataType.UNKNOWN, ThumbnailDataType.STRING, ThumbnailDataType.FILE,
+        ThumbnailDataType.BITMAP, ThumbnailDataType.LINK})
+@Retention(RetentionPolicy.SOURCE)
+public @interface ThumbnailDataType {
+    int UNKNOWN = 0;
```
Comment (veler) [inline Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/enums/ThumbnailDataType.java]: This one looks good, but please double check.

File: Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/enums/DataEntryStatus.java (hunk 1 above)
Comment (veler) [reply Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/enums/DataEntryStatus.java]: The reason why is that the status is retrieves by the app on other device and is used to defines whether a data must be keep, deleted or else after a synchronization.

**Comment by vmichalak** [comment]
It's done, the windows version is updated now :)

**Pull request merged**